src/tools/clippy/clippy_lints/src/eta_reduction.rs

```diff
@@ -203,7 +204,7 @@ fn check_clousure<'tcx>(cx: &LateContext<'tcx>, outer_receiver: Option<&Expr<'tc
                             // 'cuz currently nothing changes after deleting this check.
                             local_used_in(cx, l, args) || local_used_after_expr(cx, l, expr)
                         }) {
-                            match cx.tcx.infer_ctxt().build().err_ctxt().type_implements_fn_trait(
+                            match cx.tcx.infer_ctxt().build(TypingMode::from_param_env(cx.param_env)).err_ctxt().type_implements_fn_trait(
```

Comment:
Same as all the other lints above, I'd like to see `LateLintCtxt::typing_mode` method added so users can just do `infcx.build(cx.typing_mode())` so they dont need to think about it here.